modules/microshift-custom-ca-proc.adoc

```diff
@@ -0,0 +1,227 @@
+====
+
+.Prerequisites
+* The OpenShift CLI (`oc`) is installed.
+* You have access to the cluster as a user with the cluster administration role.
+* Certificate authorities are set up.
+* A {microshift-short} `/etc/microshift/config.yaml` configuration file exists.
+
+[IMPORTANT]
+====
+Either replace the `certificate-authority-data` in the external `kubeconfig` with the new `rootCA`, or add the parameter to the trust root of the operating system. Do not do both.
+====
+
+.Procedure
+
+. For each custom CA that you need, add an `apiServer` section called `namedCertificates` to the `/etc/microshift/config.yaml` {microshift-short} configuration file by using the following example:
++
+[source,yaml]
+----
+apiServer:
+  namedCertificates:
+   - certPath: ~/certs/api_fqdn_1.crt <1>
+     keyPath:  ~/certs/api_fqdn_1.key <2>
+   - certPath: ~/certs/api_fqdn_2.crt
+     keyPath:  ~/certs/api_fqdn_2.key
+     names: <3>
+     - api_fqdn_1
+     - *.apps.external.com
+----
+<1> Add the full path to the certificate.
+<2> Add the full path to the certificate key.
+<3> Optional. Add a list of explicit DNS names. Leading wildcards are allowed. If no names are provided, the implicit names are extracted from the certificates.
+
+. Restart the {microshift-service} to apply the CAs by running the following command:
++
+[source,terminal]
+----
+$ systemctl microshift restart
+----
+
+. Wait five minutes for system restart and apply the custom server.
+
+. Specify the `kubeconfig` to use by running the following command:
++
+[source,terminal]
+----
+$ export KUBECONFIG=/var/lib/microshift/resources/kubeadmin/<custom_cert_fqdn>/kubeconfig
+----
+
+. Check that the certificates are applied by using the following command:
++
+[source,terminal]
+----
+$ oc --certificate-authority ~/certs/ca.ca get node
+----
++
+.Example output
+[source,terminal]
+----
+oc get node
+NAME                             STATUS   ROLES                         AGE   VERSION
+dhcp-1-235-195.arm.example.com   Ready    control-plane,master,worker   76m   v1.29.2
+----
+
+. Check that the `openshift-service-ca` is running by entering the following command:
++
+[source,terminal]
+----
+$ oc --insecure-skip-tls-verify get pod -A
+----
++
+.Example output
+[source,terminal]
+----
+oc get pod -A
+NAMESPACE                              NAME                                       READY   STATUS    RESTARTS      AGE
+kube-system                            csi-snapshot-controller-5bdbd484f4-c6prb   1/1     Running   0             22m
+kube-system                            csi-snapshot-webhook-957f6d687-vb7fl       1/1     Running   0             22m
+openshift-dns                          dns-default-gbkx8                          2/2     Running   0             22m
+openshift-dns                          node-resolver-lqd7d                        1/1     Running   0             22m
```

Comment:
I dont understand where  `openshift-service-ca` comes from, why do we need to have this section? Microshift doesnt use operators to manage its CAs, its being handled internally by Microshift  binary.

Reply:
This came from the Polarion. I didn't know where the line was between our testing v. user needed to do. We can remove.

Reply:
Looks like it was there because using the `oc get node` command gives the error `Unable to connect to the server: tls: failed to verify certificate: x509: certificate signed by unknown authority` is expected. Using `--insecure-skip-tls-verify` resolves the error.